From 0dd5fd692f79cf9511ffd45f84ef992fb8bc0d70 Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@intel.com>
Date: Fri, 20 Mar 2020 21:16:24 +0000
Subject: [PATCH] curl-functions.m4: remove inappropriate AC_REQUIRE

AC_REQUIRE means "if this macro hasn't been executed already, execute it".  So in a wrapper around AC_RUN_IFELSE, AC_REQUIRE(AC_RUN_IFELSE) isn't correct at that will execute AC_RUN_IFELSE without any arguments.

With autoconf 2.69 this is basically a no-op, but with autoconf 2.70, AC_RUN_IFELSE without a default value when cross-compiling is fatal. The result is that curl with autoconf 2.70 cannot cross-compile.
---
 m4/curl-functions.m4 | 2 --
 1 file changed, 2 deletions(-)

diff --git a/m4/curl-functions.m4 b/m4/curl-functions.m4
index 9020f3942..038bf02be 100644
--- a/m4/curl-functions.m4
+++ b/m4/curl-functions.m4
@@ -7289,8 +7289,6 @@ dnl CURL_LIBRARY_PATH variable. It keeps the LD_LIBRARY_PATH
 dnl changes contained within this macro.
 
 AC_DEFUN([CURL_RUN_IFELSE], [
-   AC_REQUIRE([AC_RUN_IFELSE])dnl
-
    old=$LD_LIBRARY_PATH
    LD_LIBRARY_PATH=$CURL_LIBRARY_PATH:$old
    export LD_LIBRARY_PATH
-- 
2.34.0

